lambdas/check-existing-identity/src/main/java/uk/gov/di/ipv/core/checkexistingidentity/CheckExistingIdentityHandler.java

```diff
@@ -929,4 +934,20 @@ private void sendProfileMatchedAuditEvent(
                         new AuditRestrictedDeviceInformation(deviceInformation));
         auditService.sendAuditEvent(auditEvent);
     }
+
+    private boolean isReprovingWithF2f(CriResponseItem f2fRequest) {
+        return f2fRequest != null && f2fRequest.isReproveIdentity();
+    }
+
+    private void removeReproveIdentityFlag(boolean isF2FComplete, CriResponseItem f2fResponseItem) {
+        if (isF2FComplete && f2fResponseItem.isReproveIdentity()) {
```

Comment:
Rather than removing the flag like this, I did consider adding in a new state to the journey map to reset the session, including the CRI response like we do elsewhere. This proved difficult though due to requiring the VCs to still be in the session store for a reuse journey.
If we wanted to do this we could add a new reset type that purely resets the CRI response. This approach also adds more complexity to the journey maps.
This way seemed easier and faster - but happy to be corrected.

Reply:
What's the scenario you're trying to avoid by doing this? I guess the inverse risk is that if something blows up afterwards (before the intervention is cleared) then they'll have to start again.

If we add a check for `vcs.isPendingEvcsIdentity()` when skipping, will that help avoid the issue? I.e. so that it only skips if the identity is still pending.

Reply:
Do you mean the `isF2FComplete` flag. I feel like I did have a good reason before the weekend, but now I can't for the life of me remember what it was...

As this change checks that `isReproveIdentity` flag on the response item, I'm not sure if also checking `isPendingIdentity` is needed. Although it's made we wonder if we could just check that rather than updating the response item...

Reply:
No - because we want to know if the journey was _started_ in response to an intervention - not if the user happens to come back to core with an pre-existing pending response and a new intervention.

Reply:
Yeah exactly :)

I think maybe the scenario that you're worried about is like:
- User gets account intervention
- User reproves identity using F2F
- User's intervention is cleared
- User gets _another_ intervention
- The pending record from the first reprove is still hanging around (with `isReproveIdentity=true`), and the intervention gets immediately cleared

I don't think it's a massive risk tbh, but nice to solve if we can.

My theory with adding `isPendingEvcsIdentity` is that it'll mean you only get to complete the intervention if you're _actually_ returning from F2F (not that you just happen to have a response record kicking around).